codeworld-compiler/src/Compile.hs

```diff
@@ -0,0 +1,156 @@
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+{-
+  Copyright 2017 The CodeWorld Authors. All rights reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-}
+
+module Compile (
+
+    compileSource
+
+)
+where
+
+import           Control.Concurrent
+import           Control.Monad
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString as B
+import           System.Directory
+import           System.FilePath
+import           System.IO
+import           System.IO.Temp (withSystemTempDirectory)
+import           System.Process
+import           Text.Regex.TDFA
+
+import ErrorSanitizer
+
+compileSource :: FilePath -> FilePath -> FilePath -> [String] -> IO Bool
```

Comment:
In order to make it a standalone compiler it is more likely that one can also pass the list of arguments apart from just passing the args as (haskell, or codeworld). making it more open for other options too.
If that part is not of much importance now then everything can be easily shifted back from [String] thing to just String.